### DISCUSSION TITLE
1.6.0

### DISCUSSION THREAD

**Discussion by ruibaby** [halo-dev/halo#2544] [General]
## Features

- SMTP 添加 STARTTLS 的设置支持。 halo-dev/halo#1861 halo-dev/console#552 @ntdgy @wangxiaoerYah
- 为邮件通知模板添加更多的可用参数，`email`、`status`、`createTime`、`authorUrl`。 halo-dev/halo#2095 @Yhcrown @iRoZhi
- 后台附件设置中添加 `去除图片 EXIF 信息` 的选项。 halo-dev/halo#2168 halo-dev/console#554 @SladeGranger @eziosudo @52lemon6
- 后台文章标签管理支持清理未使用标签。 halo-dev/console#587 @ruibaby
- 重构后台的友情链接管理页面，现在支持拖动排序以及分组管理。 halo-dev/console#574 halo-dev/halo#2105 @Camsyn @daifiyum @gungnir479

## Improvements

- 优化文章编辑时间更新的逻辑，目前更改为仅在修改文章标题或者内容时才更新。 halo-dev/halo#2195 @wxyShine @ListenV
- 修改后台管理页面为直接渲染而不是重定向，以更好地兼容反向代理。 halo-dev/halo#2259 @viticis
- 后台在构建时提供 gzip 压缩的资源。 halo-dev/console#547 @2211898719
- 后台主题设置界面顶部提供了保存设置的按钮。 halo-dev/console#549 @Aanko
- 修改后台页面的 lang 属性，由 `zh-cmn-Hans` 改为 `cmn-Han`，`zh-cmn-Hans` 已废弃。 halo-dev/console#576 @wordlesswind
- 优化后台附件库列表以及选择附件弹框的布局。 halo-dev/console#580 @ruibaby
- 修改后台底部的 `Power by Halo` 为 `Powered by Halo`。 halo-dev/console#597 @liaocp666
- 优化后台日志管理设置公开/隐藏状态的方式。 halo-dev/console#610 @zjy4fun @manction
- 优化后台在个人资料中设置头像的逻辑，现在无需再提交个人资料表单。 halo-dev/console#619 @wxyShine
- 优化后台图库批量从附件添加时，按照选择顺序倒序排列，即最先选择的图片在最前面。 halo-dev/console#631 @zjy4fun @zyy247796143

## Bug Fixes

- 修复文章详情页 `meta_description` 为空的问题。 halo-dev/halo#2282 @guqing @ruibaby
- 修复批量删除外部云存储的时候，因为文件不存在导致的删除失败问题。 halo-dev/halo#2317 @JustinLiang522 @129duckflew
- 修复了七牛云存储附件无法上传非图片文件的问题。 halo-dev/halo#2331 @AirboZH @hexWars
- 修复未审核评论回复会发送邮件提醒的问题。 halo-dev/halo#2340 @AirboZH @cetr
- 修复在单个分类所属文章页面页面变量中无法获取子分类中文章的问题。 halo-dev/halo#2405 @JustinLiang522 @hugeleaf
- 修复管理员资料表单中邮件地址的字符数限制。 halo-dev/console#571 @Yhcrown
- 修复了无法设置日志、相册、链接页标题和每页显示条数的问题。 halo-dev/console#601 @JustinLiang522 @manction

## Dependencies

- Docker 镜像的基础镜像使用 Eclipse Temurin 镜像替代 AdoptOpenJDK。 halo-dev/halo#2120 @wordlesswind

**Full Changelog**: https://github.com/halo-dev/halo/compare/v1.5.2...v1.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/halo-dev/halo/releases/tag/v1.6.0'>1.6.0</a>.</em>

**Comment by itzsh** [comment]
从1.5.4升级到1.6.0时失败，但是升级到1.5.5成功

**Reply by JohnNiang** [reply]
你可能需要提供更多的信息让大家帮助到你：

1. 升级失败的具体错误
2. 升级的详细步骤，尽量能够稳定复现
